staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -117,7 +117,11 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope *RequestSc
 		trace.Step("About to check admission control")
 		if admit != nil && admit.Handles(admission.Delete) {
 			userInfo, _ := request.UserFrom(ctx)
-			attrs := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, dryrun.IsDryRun(options.DryRun), userInfo)
+			// For backwards compatibility, we don't change the options TypeMeta that are passed to rest storage Delete.
```

Comment:
I'm trying to think of of what problem just using meta.k8s.io/v1 DeleteOptions would cause... did you run into issues  when you tried to set this to meta.k8s.io/v1?

Reply:
by the time we get here, I thought whatever group/version the user provided was consumed and the options object was just a &metav1.DeleteOptions{} struct with no typemeta. Is that not the case?

Reply:
I've reviewed the downstream code and you're right- there is no need to retain the TypeMeta provided by the user. The only thing we need to do to remain backward compatibility is to pass validation if a client submits per group DeleteOptions. I was being overly conservative ;)